### PR TITLE
MEL-385: Deprecate imms module

### PR DESCRIPTION
**Pull request by kael89**
### Issue #:
MEL-385

### CODE REVIEW

File: packages/database/src/migrations/20210813012332-DeleteImmsProjectAndVizes-modifies-data.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    `SELECT * FROM dashboard_item WHERE code ILIKE 'Imms_%';`,
+  );
+  const immsDashboardItemIds = immsDashboardItems.map(di => di.id);
+  await deleteDashboardItemsByIds(db, immsDashboardItemIds);
+
+  await deleteMapOverlayGroups(db, IMMS_MAP_OVERLAY_GROUPS);
+  const { rows: immsMapOverlays } = await db.runSql(`
+      SELECT * FROM "mapOverlay"
+      WHERE "projectCodes" IN ('{imms,explore}')`);
+  const immsMapOverlayIds = immsMapOverlays.map(di => di.id);
```
Comment (chris-bes) [inline packages/database/src/migrations/20210813012332-DeleteImmsProjectAndVizes-modifies-data.js:113]: I noticed there's also some other map overlays that has `imms` in their project codes, but used in other projects as well. So maybe we want to remove `imms` from their projectCodes as well?

Query:
```
SELECT * FROM "mapOverlay"
WHERE "projectCodes"::text like '%imms%'
```

File: packages/web-frontend/cypress/config/dashboardReportUrls/default.json [hunk 2]
```diff
@@ -199,32 +199,6 @@
     "orgUnit": "FETP-DYO3",
     "dashboard": "FETP"
   },
-  {
```
Comment (chris-bes) [inline packages/web-frontend/cypress/config/dashboardReportUrls/default.json:202]: The most satisfying PR I have reviewed ever 😆

File: packages/database/src/migrations/20210813012332-DeleteImmsProjectAndVizes-modifies-data.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+const toggleEntityTriggers = async (db, enable) => {
+  const action = enable ? 'ENABLE' : 'DISABLE';
+
+  await db.runSql(`ALTER TABLE entity_relation ${action} TRIGGER entity_relation_trigger`);
+  await db.runSql(`ALTER TABLE entity_hierarchy ${action} TRIGGER entity_hierarchy_trigger`);
+  await db.runSql(`ALTER TABLE entity ${action} TRIGGER entity_trigger`);
+};
+
+exports.up = async function (db) {
+  await deleteDashboards(db, IMMS_DASHBOARDS);
```
Comment (chris-bes) [inline packages/database/src/migrations/20210813012332-DeleteImmsProjectAndVizes-modifies-data.js:102]: Same as the map overlays, I think there are some generic dashboard reports that are used in multiple projects, and some of them have `imms` in their project codes. So perhaps we also want to remove `imms` from `project_codes` (dashboard_relation table):

Query I used:

```
select dashboard.* 
from dashboard
join dashboard_relation on dashboard.id = dashboard_relation.dashboard_id
where project_codes::text like '%imms%';
```

**Review by chris-bes — COMMENTED** [review]
LGTM! Just a minor questions about the generic dashboard reports/map overlays that are used in multiple projects (including Imms)

File: packages/database/src/migrations/20210813012332-DeleteImmsProjectAndVizes-modifies-data.js (hunk 1 above)
Comment (kael89) [reply packages/database/src/migrations/20210813012332-DeleteImmsProjectAndVizes-modifies-data.js:113]: Oops good point!

File: packages/database/src/migrations/20210813012332-DeleteImmsProjectAndVizes-modifies-data.js (hunk 3 above)
Comment (kael89) [reply packages/database/src/migrations/20210813012332-DeleteImmsProjectAndVizes-modifies-data.js:102]: Great catch!~

**Comment by kael89** [comment]
Ready to re-review @chris-bes, see the last 2 commits. The first adds a function to refresh sort order for projects, filling the gap left by `imms`

File: packages/database/src/migrations/20210813012332-DeleteImmsProjectAndVizes-modifies-data.js [hunk 4]
```diff
@@ -98,6 +98,16 @@ const toggleEntityTriggers = async (db, enable) => {
...
 
+const refreshProjectSortOrder = async db => {
+  const { rows: projects } = await db.runSql(`
+    SELECT * FROM project WHERE sort_order IS NOT NULL
+    ORDER BY sort_order ASC`);
+
+  for (const [i, project] of projects.entries()) {
+    await db.runSql(`UPDATE PROJECT SET sort_order = ${i} WHERE id = '${project.id}'`);
+  }
+};
```
Comment (chris-bes) [inline packages/database/src/migrations/20210813012332-DeleteImmsProjectAndVizes-modifies-data.js:114]: Ah nice catch!

**Review by chris-bes — APPROVED** [review]
LGTM!

**Comment by kael89** [comment]
@chris-bes I pushed a commit to address https://linear.app/tupaia/issue/MEL-385#comment-be2229e3

**Review by chris-bes — APPROVED** [review]

**Pull request merged**